File: c/shift-register/shift-1pin.c

#include <wiringPi.h>

void pulse (int pin, double t);

// shift register data signal
int SHIFT = 7;

// sleep timing
int ONE = 1;
int ZERO = 15;
int PAUSE = 50;



//testing output
int main (void){
  wiringPiSetup () ;
  pinMode (SHIFT, OUTPUT) ;
  digitalWrite (SHIFT, HIGH) ;
  //delay (ZERO);
  //digitalWrite (SHIFT, LOW);  
  //delay (ZERO);
  //digitalWrite(SHIFT, HIGH);
  for (;;)  {
  //for (int x = 0; x < 8; x++){
    pulse_low(SHIFT, ONE);
    delayMicroseconds (PAUSE) ;
    pulse_low(SHIFT, ZERO);
  }

}

// pulse function to pulse a pin
// pin - GPIO pin to pulse
// t - pulse delay between high and low
void pulse_low(int pin, int t){
  digitalWrite (pin, LOW);
  //printf("low");
  //delay( t);
  delayMicroseconds (t) ;
  digitalWrite (pin, HIGH) ; 
}

File: python/shift-register/shift-1pin.py
import RPi.GPIO as gpio
from time import sleep 
gpio.setmode(gpio.BCM)

# shift pin
SHIFT = 26

gpio.setup(SHIFT, gpio.OUT)
gpio.output(SHIFT, gpio.HIGH)

ONE = 1
ZERO = 15
PAUSE = 50

def pulse_low(pin, t):
	gpio.output(pin, 0)
    sleepM
    gpio.output(pin, 1)
	
	
	return

try:
	while True:
        pulse(SHIFT, )


except KeyboardInterrupt:
	gpio.cleanup()
